jpo-asn-runtime/src/main/java/us/dot/its/jpo/asn/runtime/types/Asn1SequenceOf.java

```diff
@@ -0,0 +1,40 @@
+package us.dot.its.jpo.asn.runtime.types;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import java.util.ArrayList;
+
+
+public abstract class Asn1SequenceOf<T extends Asn1Type>
+    extends ArrayList<T>
+    implements Asn1Type {
+
+    final Class<T> itemClass;
+    final long sizeLowerBound;
+    final long sizeUpperBound;
+
+    public Asn1SequenceOf(Class<T> itemClass, long sizeLowerBound, long sizeUpperBound) {
+        this.itemClass = itemClass;
+        this.sizeLowerBound = sizeLowerBound;
+        this.sizeUpperBound = sizeUpperBound;
+    }
+
+    @JsonIgnore
+    public Class<T> getItemClass() {
+        return itemClass;
+    }
+
+    @JsonIgnore
+    public long getSizeLowerBound() {
+        return sizeLowerBound;
+    }
+
+    @JsonIgnore
+    public long getSizeUpperBound() {
+        return sizeUpperBound;
+    }
+
+    @SuppressWarnings("unchecked")
+    public boolean add(Asn1Type item) {
+        return super.add((T)item);
+    }

```

Comment:
chore: there isn't a need to do an unchecked cast if we use the template type as the method param. i've made this change locally and confirmed all tests pass
```suggestion
    public boolean add(T item) {
        return super.add(item);
    }
```

Reply:
This change doesn't work in this code in the example generator, which will need to be able to create an Asn1SequenceOf via reflection without knowing the type of items in it.  That unchecked cast is not pretty, I agree, but it was the easiest way to get this to do what I wanted, considering we have to deal with Java Generics type erasure.  

https://github.com/CDOT-CV/jpo-asn-pojos/blob/j2735-2024/jpo-asn-test-generator/src/main/java/us/dot/its/jpo/asn/testgenerator/SequenceOfGenerator.java

![image](https://github.com/user-attachments/assets/fb89d872-4361-4ec2-aaba-b492d7115a23)